omnigibson/prims/entity_prim.py

```diff
@@ -1531,18 +1567,28 @@ def create_attachment_point_link(self):
     def _dump_state(self):
         # We don't call super, instead, this state is simply the root link state and all joint states
         state = dict(root_link=self.root_link._dump_state())
-        joint_state = dict()
-        for prim_name, prim in self._joints.items():
-            joint_state[prim_name] = prim._dump_state()
-        state["joints"] = joint_state
+        if self.n_joints > 0:
+            state["joint_pos"] = self.get_joint_positions()
+            state["joint_vel"] = self.get_joint_velocities()
+            state["joint_eff"] = self.get_joint_efforts()
+            # state["joint_pos_target"] = self.get_joint_position_targets()
+            # state["joint_vel_target"] = self.get_joint_velocity_targets()
 
         return state
 
     def _load_state(self, state):
         # Load base link state and joint states
         self.root_link._load_state(state=state["root_link"])
-        for joint_name, joint_state in state["joints"].items():
-            self._joints[joint_name]._load_state(state=joint_state)
+
+        # TODO: Remove this backwards-compatible part once we re-sample scenes
+        if "joints" in state:
+            for joint_name, joint_state in state["joints"].items():
+                self._joints[joint_name]._load_state(state=joint_state)
+
+        elif self.n_joints > 0:
+            self.set_joint_positions(state["joint_pos"])
+            self.set_joint_velocities(state["joint_vel"])
+            self.set_joint_efforts(state["joint_eff"])
```

Comment:
Is there any benefit of saving/loading joint states at the entity prim level rather than the joint level?